pkg/store/zookeeper.go

```diff
@@ -131,9 +131,13 @@ func (s *Zookeeper) Watch(key string, stopCh <-chan struct{}) (<-chan *KVPair, e
 	go func() {
 		defer close(watchCh)
 
-		// GetW returns the current value before setting the watch.
-		watchCh <- &KVPair{key, resp, uint64(meta.Version)}
+		// Get returns the current value before setting the watch.
+		watchCh <- pair
 		for {
+			_, _, eventCh, err := s.client.GetW(fkey)
```

Comment:
It's a one time watch thus the necessity to initialize it at each turn (If you want to keep watching events on that key).